drivers/pinctrl/pinctrl_ite_it8xxx2.c

```diff
@@ -308,6 +326,16 @@ static int pinctrl_it8xxx2_init(const struct device *dev)
 	 * have to set UART1PSEL = 1 in UART1PMR register.
 	 */
 
+#ifdef CONFIG_SOC_IT8XXX2_REG_SET_V2
+	/*
+	 * Swap the default I2C2 SMCLK2/SMDAT2 pins from GPC7/GPD0 to GPF6/GPF7,
```

Comment:
Should this be optional? It seems odd to unconditionally remap some of the IOs.

Reply:
Yes, that is correct. 

I have created these mapped pins to the it8xxx2-pinctrl-map.dtsi file.
Added i2c2_clk_gpc7_default/i2c2_data_gpd0_default, 
            i2c3_clk_gpb2_default/i2c3_data_gpb5_default
            i2c5_clk_gpe1_default/i2c5_data_gpe2_default

If the user selects GPC7/GPD0, the as configuration below will select IT8XXX2_ALT_FUNC_4.
And the `pinctrl_gpio_it8xxx2_configure_pins` function in pinctrl_ite_it8xxx2.c will make the corresponding set.
The `IT8XXX2_GPIO_SMB2PS `bit will get set.
```
	i2c2_clk_gpc7_default: i2c2_clk_gpc7_default {
		pinmuxs = <&pinctrlc 7 IT8XXX2_ALT_FUNC_4>;
	};
	i2c2_data_gpd0_default: i2c2_data_gpd0_default {
		pinmuxs = <&pinctrld 0 IT8XXX2_ALT_FUNC_4>;
	};
```

Reply:
Thank you for the explanation.